src/components/tooltip/GCMarker.js

```diff
@@ -423,80 +430,104 @@ function _treeInsert(
   let node = tree.get(component);
   if (!node) {
     // Make a new node and grow the tree in this direction.
-    node = { branches: new Map() };
+    const value = { name: '(temp)', time: 0 };
+    node = { branches: new Map(), leaf: false, value };
 
     tree.set(component, node);
   }
 
   if (path.length > 0) {
     // There are more path components.  This node should be a branch if it
     // isn't one already.
-    if (node.value) {
-      // We delete the value to change this node from a leaf to a branch.
-      delete node.value;
-    }
+    node.leaf = false;
     _treeInsert(node.branches, path, phase);
   } else {
     // Make the new node leaf node.
-    if (node.value) {
+    if (node.value.name !== '(temp)') {
       console.error(
-        'Duplicate phases in _treeInsert in MarkerTooltipContents.js'
+        `Duplicate phase ${phase.name} in _treeInsert in MarkerTooltipContents.js`
       );
       return;
     }
     node.value = phase;
+    node.leaf = true;
   }
 }
 
-function _treeGetLeaves(
-  tree: Map<string, PhaseTreeNode>
-): Array<PhaseTimeTuple> {
-  const leaves = [];
+function _treeMapNodes<T>(
+  tree: Map<string, PhaseTreeNode>,
+  f: (node: PhaseTreeNode) => T | null,
```

Comment:
For the filter function, we can use the name `filter` instead of just `f`, to better indicate what we'll do with that.
For the loop function, `f` is fine, probably.